packages/expo/bundledNativeModules.json

```diff
@@ -76,7 +76,7 @@
   "expo-branch": "~2.0.0",
   "react-native-appearance": "~0.3.1",
   "react-native-safe-area-context": "0.6.0",
-  "react-native-shared-element": "~0.5.1",
+  "react-native-shared-element": "~0.5.5",
```

Comment:
we might want to follow a similar versioning scheme for `react-native-shared-element` as we do for `react-native-gesture-handler` and other pkgs:

patch version - backwards compatible with native on the same minor version, includes js changes only (fixes or api additions)
minor - includes native and/or js changes (fixes or api additions)
major - breaking api changes

it's not exactly the definition of semver but it works well for us.

until we start following that `react-native-shared-element` we should probably change the dep to `"0.5.5"` without the `~`

Reply:
I hear what you'r saying. Is there a version scheme policy somewhere that I can view or link to in my project (if I would want to adopt it)?

I'm not entirely sure this will work for the packages themselves though. I understand it's useful for Expo and in managing its package versions though. Regarding `react-native-shared-element`, most of it's code is native (> 95%), so most of the time the changes will be in there. I do think that "patches" can (and should be able to) occur in both native and JS code (as long as they don't break any APIs).

Thinking about it, I would rather opt for a slightly different scheme, which I feel is more accurate in describing "breaking" changes.

```
patch - backwards compatible with native & js. Does not introduce any API changes to native, but may include additions on the JS side)
minor - introduces changes the native API and/or fixes/additions in JS
major - breaking API changes (JS)
```

Also looking at `react-native-gesture-handler`, the [last release (1.5.6)](https://github.com/software-mansion/react-native-gesture-handler/releases) bumps the patch version, but the changes are in the native code, so it doesn't seem to follow the scheme that you mentioned.

Reply:
> until we start following that react-native-shared-element we should probably change the dep to "0.5.5" without the ~

Good one, I've changed it 👍

Reply:
I think your proposal is totally sensible. Pinged @kmagiera!

Reply:
We usually stick to this convention despite it not being documented anywhere. I believe version 1.5.4 was the only patch release that include a native change and this was due to a human error.

We are working on partially automating release process now to avoid such mistakes in future.

As for following the convention only for cases that does not change the API surface I don't think it makes any difference from the user perspective but makes it more difficult to manage for the maintainers and here is why:
1. It is much easier to automatically check for native code changes and require minor bump
2. People who use expo are stuck on a certain native release, it does not make any difference to them when they install newer version if that version does not have any changes relevant to them (only native code changes). Given they use different code on the native side we may have problems with people reporting issues for wrong version. There are also cases where behavior can change despite no API surface updates in which case people may rely on a mismatched documentation (for a different version than they actually run)

Reply:
> We usually stick to this convention despite it not being documented anywhere. I believe version 1.5.4 was the only patch release that include a native change in the past year or so, and this was due to a human error.

Okay I see 👍 



> As for following the convention only for cases that does not change the API surface I don't think it makes any difference from the user perspective but makes it more difficult to manage for the maintainers and here is why:
>
> 1. It is much easier to automatically check for native code changes and require minor bump
>
> 2. People who use expo are stuck on a certain native release, it does not make any difference to them when they install newer version if that version does not have any changes relevant to them (only native code changes). Given they use different code on the native side we may have problems with people reporting issues for wrong version. There are also cases where behavior can change despite no API surface updates in which case people may rely on a mismatched documentation (for a different version than they actually run)

Okay I think I understand the general gist here. Like Brent said, it's not really semver but I understand how this makes release management simpler.

I'll adopt the same strategy in the next release of react-native-shared-element. 👍